utility/database/check_can_associate_unique_ids.go

```diff
@@ -0,0 +1,61 @@
+package utility_database
+
+import (
+	"slices"
+
+	"github.com/PretendoNetwork/nex-go/v2"
+	"github.com/PretendoNetwork/nex-go/v2/types"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/v2/globals"
+)
+
+// This handles all relevant validity checking (including the existence of an id, if the password matches, etc)
+func CheckCanAssociateUniqueIDs(manager *common_globals.UtilityManager, userPid types.PID, uniqueIds, passwords types.List[types.UInt64]) *nex.Error {
+	var uniqueId, dbPassword types.UInt64
+	var associatedPid types.PID
+
+	rows, err := manager.Database.Query(`SELECT unique_id, associated_pid, password FROM utility.unique_ids WHERE unique_id=ANY($1)`,
+		uniqueIds,
+	)
+	if err != nil {
+		return nex.NewError(nex.ResultCodes.Core.Unknown, err.Error())
+	}
+
+	rowCount := 0
+	for rows.Next() {
+		rowCount++
+
+		err = rows.Scan(
+			&uniqueId,
+			&associatedPid,
+			&dbPassword,
+		)
+		if err != nil {
+			return nex.NewError(nex.ResultCodes.Core.Unknown, err.Error())
+		}
+
+		// TODO - Is this a correct assumption?
+		if associatedPid != userPid {
+			return nex.NewError(nex.ResultCodes.Core.Unknown, "One of the unique ids is already owned by this user")
+		}
+
+		targetConnection := manager.Endpoint.FindConnectionByPID(uint64(associatedPid))
+		if targetConnection != nil || !manager.AllowUniqueIDStealing {
```

Comment:
```suggestion
		if targetConnection != nil && !manager.AllowUniqueIDStealing {
```

Otherwise this will always error

Reply:
It shouldn't
As of latest commit, when unique id stealing is enabled, if the user is not the owner, and the owner is online, it will error
When unique id stealing is disabled, and the user is not the owner, it will error